frontend/lib/parsing/parsing-queries.cpp

```diff
@@ -100,25 +102,226 @@ static Parser helpMakeParser(Context* context,
   }
 }
 
+// <7F>HPECHPL
+#define LIBRARY_MAGIC (uint64_t)0x4C5048434550487F
+#define LIBRARY_VERSION_MAJOR 0
+#define LIBRARY_VERSION_MINOR 1
+
+//
+// The library file format (whitespace not significant):
+// <magic number, uint64_t>
+// <library version, major, int>
+// <library version, minor, int>
+// <chpl version, major, int>
+// <chpl version, minor, int>
+// <chpl version, update, int>
+//
+// <user/std module descriptor, std::string>
+//
+// N:<number of BuilderResult entries, uint64_t>
+//   0..N-1: <file path i, std::string><offset i, uint64_t>
```

Comment:
Do I interpret this "Nth file path is a std::string" as the `std::string` buffer being embedded inline? Or is the `<offset i, uint64_t>` indicating the offset into the string pool? Might work the format wording a bit to indicate that. Do we have to use `std::string` for the type or can we just use `const char*`?

Reply:
It was my intention to indicate that there were ``N`` entries consisting of two parts: an ``std::string`` representing the file path for this entry, and a ``uint64_t`` representing the offset to that file path's builder result.

I went with a ``std::string`` here because in the ``const char*`` case I would just serialize the length as well, just as we do with ``std::string``.